src/boot/lib/ast.ml

```diff
@@ -346,8 +348,10 @@ and ty =
   | TyRecord of info * ty Record.t * ustring list
   (* Variant type *)
   | TyVariant of info * (ustring * Symb.t) list
+  (* Type constructors *)
+  | TyCon of info * ustring * Symb.t
```

Comment:
Right, I just kept the machinery which was already there, but I don't think it's needed. For the record, the parsing functions in `bootparser` actually symbolize the AST, but then discard the symbols.

Reply:
It would be good to give a type alias for `bool` in the `TmVar` case. Otherwise, it is hard to know what this boolean actually means (without trying to derive it from the code)